### PR TITLE
refactor: Replace `StringOptions` with `ProtectedString` for encoder/decoder schema `type` fields

### PR DESCRIPTION
**Pull request by ksbrar**

### CODE REVIEW

File: ludwig/schema/decoders/base.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ class PassthroughDecoderConfig(BaseDecoderConfig):
 class RegressorConfig(BaseDecoderConfig):
     """RegressorConfig is a dataclass that configures the parameters used for a regressor decoder."""
 
-    type: str = schema_utils.StringOptions(
+    type: str = schema_utils.ProtectedString(
         ["regressor"],
         default="regressor",
         allow_none=False,
```
Comment (arnavgarg1) [inline ludwig/schema/decoders/base.py:82]: @ksbrar I think this should likely be updated to

```
type: str = schema_utils.ProtectedString("regressor", description="...")
```

Comment (ksbrar) [reply ludwig/schema/decoders/base.py:82]: woops :)

File: ludwig/schema/encoders/sequence_encoders.py [hunk 2]
```diff
@@ -47,8 +45,8 @@ class SequencePassthroughConfig(BaseEncoderConfig):
 @dataclass(repr=False)
 class SequenceEmbedConfig(BaseEncoderConfig):
 
-    type: str = schema_utils.StringOptions(
-        ["embed"],
+    type: str = schema_utils.ProtectedString(
+        "embed",
```
[the remark this answers is not in the document]
Comment (ksbrar) [reply ludwig/schema/encoders/sequence_encoders.py:49]: woops

**Review by arnavgarg1 — APPROVED** [review]
Look good! Not sure why tests are failing but I'd just re-run/make sure they pass before merging even though this is a pretty harmless change

**Review by connor-mccorm — APPROVED** [review]

**Pull request merged**